hax/hax/ha/events.py

```diff
@@ -154,12 +169,30 @@ def ack(self):
         self.consumer.ack()
 
 
-class EventProducer():
-    """
-    Module responsible for publishing the event to message bus
-    """
+class Node():
```

Comment:
I think this entire ha module needs a bit of redesigning, for instance,
1. Message bus interface must be separated out from ha module (MessageInterface) and instantiated in hax.py.
2. There needs to be a HA class that takes MessageInterface as a parameter.
3. Message bus specific Consumers and Producers must be encapsulated inside MessageInterface.
4. Define `HaEvent` interface class, with methods as send(), recv(), etc.
5. Define various ha event class based on the topics, these classes will be inherited from `HaEvent` class and implement send() and receive() functions. Respective Send functions will send over corresponding MessageInterface or post he event locally to local Ha module in future.
4. HA class will include list/dictionary of such ha event types created.
5. when ha thread is created, HA class will be passed as a parameter.
6. Define ha Object types, e.g. Node, Cluster, drive, etc. We need to map the respective motr fids to their corresponding global object ids. Respective utils function must be defined to map motr fid to ha object fid and create event.
7. Create a ha event corresponding to the object type though a generic create_event() -> HaEvent, taking ha object type as an input. And then the HaEvent must be posted to the respective event type classes based on the topics, local or remote. 
Such an abstraction will help us easily add and remove certain parts of the functionality.